src/extension.ts

```diff
@@ -202,11 +209,37 @@ function onShown() {
 }
 
 function onHide() {
+  // Restore sidebar if it was configured to be auto-hidden
+  if (globalConfig.autoHideSideBar && globalConfig.autoRestoreSideBar) {
```

Comment:
This basically ties these two settings together, in order to restore the sidebar, you have to have it set to be hidden...happy to change this to something else if you think it makes more sense to have it be standalone. E.g. always attempt to restore the sidebar regardless or any other setting or UI state.

Same goes for the other conditionals below, too.